File: app/src/main/java/com/felipemiranda/desafioconcrete/network/api/SearchApi.java
package com.felipemiranda.desafioconcrete.network.api;


import com.felipemiranda.desafioconcrete.model.ItemDetail;
import com.felipemiranda.desafioconcrete.model.response.SearchResponse;

import java.util.ArrayList;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.Url;

/**
 * Created by felipemiranda
 */

public interface SearchApi {

    @GET("search/repositories?q=language:Java&sort=stars")
    Call<SearchResponse> requestSearch(@Query("page") int page);

    @GET
    Call<ArrayList<ItemDetail>> requestItemDetail(@Url String url);
}

File: app/src/main/java/cl/jesualex/desafio_android/search/data/remote/SearchApi.kt
package cl.jesualex.desafio_android.search.data.remote

import cl.jesualex.desafio_android.search.data.domain.entity.SearchBase
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by jesualex on 2019-05-28.
 */
interface SearchApi {
    @GET("repositories") fun search(
        @Query("q") lang: String,
        @Query("sort") sortBy: String,
        @Query("page") page: Int
    ): Observable<SearchBase>
}